File: 121.java
class Solution 
{
    public int maxProfit(int[] prices) 
    {   
        int buy = prices[0],rt = 0;
        for(int i = 1;i<prices.length;i++)
        {
            if(buy>prices[i])
            {
                buy = prices[i];
            }
            else
            {
                rt = Math.max(rt,prices[i]-buy);
            }
        }
        return rt;
    }
}

File: 121.cpp
class Solution {
public:
  int maxProfit(vector<int> &prices) {
    int buy = prices[0];
    int maxpro = 0;
    for (int i = 1; i < prices.size(); i++) {
      if (prices[i] < buy)
        buy = prices[i];
      else {
        maxpro = max(maxpro, prices[i] - buy);
      }
    }
    return maxpro;
  }
};
